content/docs/examples/advanced-gateways/egress-mongo/index.md

```diff
@@ -0,0 +1,1111 @@
+
+1.  Refresh the web page of the application and verify that the ratings are displayed correctly.
+
+1.  Proceed to the following section.
+
+#### Mutual TLS between the sidecar proxies and the egress gateway
+
+You may want to enable [mutual TLS Authentication](/docs/tasks/security/mutual-tls/) between the sidecar proxies of
+your MongoDB clients and the egress gateway to let the egress gateway monitor the identity of the source pods and to
+enable Mixer policy enforcement based on that identity. By enabling mutual TLS you also encrypt the traffic.
+If you do not need mutual TLS, proceed to the next section.
+
+1.  Create an egress `Gateway` for your MongoDB service, and destination rules and virtual services
+    to direct the traffic through the egress gateway and from the egress gateway to the external service.
+
+    {{< text bash >}}
+    $ kubectl apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: istio-egressgateway
+    spec:
+      selector:
+        istio: egressgateway
+      servers:
+      - port:
+          number: 443
+          name: tls
+          protocol: TLS
+        hosts:
+        - $MONGODB_HOST
+        tls:
+          mode: MUTUAL
+          serverCertificate: /etc/certs/cert-chain.pem
+          privateKey: /etc/certs/key.pem
+          caCertificates: /etc/certs/root-cert.pem
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: DestinationRule
+    metadata:
+      name: egressgateway-for-mongo
+    spec:
+      host: istio-egressgateway.istio-system.svc.cluster.local
+      subsets:
+      - name: mongo
+        trafficPolicy:
+          loadBalancer:
+            simple: ROUND_ROBIN
+          portLevelSettings:
+          - port:
+              number: 443
+            tls:
+              mode: ISTIO_MUTUAL
+              sni: $MONGODB_HOST
+    ---
+    apiVersion: networking.istio.io/v1alpha3
+    kind: VirtualService
+    metadata:
+      name: direct-mongo-through-egress-gateway
+    spec:
+      hosts:
+      - $MONGODB_HOST
+      gateways:
+      - mesh
+      - istio-egressgateway
+      tcp:
+      - match:
+        - gateways:
+          - mesh
+          destinationSubnets:
+          - $MONGODB_IP/32
+          port: $MONGODB_PORT
+        route:
+        - destination:
+            host: istio-egressgateway.istio-system.svc.cluster.local
+            subset: mongo
+            port:
+              number: 443
+      - match:
+        - gateways:
+          - istio-egressgateway
+          port: 443
+        route:
+        - destination:
+            host: $MONGODB_HOST
+            port:
+              number: $MONGODB_PORT
+          weight: 100
+    EOF
+    {{< /text >}}
+
+1.  Proceed to [Verify that the TCP egress traffic is directed through the egress gateway](#verify-that-the-tcp-egress-traffic-is-directed-through-the-egress-gateway).
+
+#### Direct TCP traffic through an egress gateway without mutual TLS
```

Comment:
Maybe we should switch this one to use the new simpler external network endpoint ServiceEntry approach.

Reply:
@frankbu I am still not convinced that the endpoints approach is better. It requires specifying both the endpoints and the virtual services, while it spares the definition of the DestinationRule. In the current approach, you do not specify the endpoints, but specify a DestinationRule instead. I propose that we handle it separately for this documentation item, and for other egress gateway examples.